cmd/mdatagen/templates/metrics.go.tmpl

```diff
@@ -55,6 +55,47 @@ func DefaultMetricsSettings() MetricsSettings {
 	}
 }
 
+// ResourceAttributeSettings provides common settings for a particular metric.
+type ResourceAttributeSettings struct {
+	Enabled bool `mapstructure:"enabled"`
+
+	enabledProvidedByUser bool
+}
+
+// IsEnabledProvidedByUser returns true if `enabled` option is explicitly set in user settings to any value.
+func (ras *ResourceAttributeSettings) IsEnabledProvidedByUser() bool {
```

Comment:
This method isn't needed anymore. It was intended to be exposed for scraper authors to show warnings based on user configuration but the warnings are now built into the builder